source/ui/theme.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+* C++ Assembler Interpreter
+* Copyright (C) 2016 Chair of Computer Architecture
+* at Technical University of Munich
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <QDir>
+#include <QIODevice>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QString>
+#include <QVariantMap>
+
+#include "common/assert.hpp"
+#include "common/utility.hpp"
+#include "ui/theme.hpp"
+
+Theme* Theme::_theme = nullptr;
+
+Theme* Theme::reset(const QString& themeName, QObject* parent) {
+  assert::that(!themeName.isEmpty());
+  delete _theme;
+  return _theme = new Theme(themeName, parent);
+}
+
+Theme& Theme::instance() {
+  assert::that(_theme != nullptr);
+  return *_theme;
+}
+
+Theme::Theme(const QString& themeName, QObject* parent) : super(this, parent) {
+  assert::that(!themeName.isEmpty());
+  load(themeName);
+}
+
+void Theme::load(const QString& themeName) {
```

Comment:
There will be a fixed number of themes that the user can select from a combo box, so unless the user messes with the configuration, nothing bad should happen. But for that case, I'll add some handling. Thanks.